### PR TITLE
[MM-39738] E2E test for Open Settings modal using keyboard shortcuts

### PR DESCRIPTION
**Pull request by codedsun**
#### Summary

feat: Added e2e test for opening settings modal for shortcut
#### Ticket Link

  Fixes https://github.com/mattermost/desktop/issues/1907
#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
- [ ] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
- [ ] executed `npm run lint:js` for proper code formatting

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by codedsun** [comment]
@devinbinnie Please review, test are passing! 🎉

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks a lot @codedsun!

**Review by M-ZubairAhmed — COMMENTED** [review]
I dont know why but i dont see the ✔️ on the test when i ran it locally.
image

i changed the `settings` to arbitary value the test passed still 🤔 
what do you think @devinbinnie

**Comment by devinbinnie** [comment]
> I don't know why but i don't see the ✔️ on the test when i ran it locally. image
> 
> i changed the `settings` to arbitrary value the test passed still 🤔 what do you think @devinbinnie

The test ran just fine for me when I pulled it down. Did you maybe have an older version?
The `-` usually means the test is skipped, but that only happens if you add `.skip` to the `it` function.

**Comment by devinbinnie** [comment]
> > I don't know why but i don't see the ✔️ on the test when i ran it locally. image
> > i changed the `settings` to arbitrary value the test passed still 🤔 what do you think @devinbinnie
> 
> The test ran just fine for me when I pulled it down. Did you maybe have an older version? The `-` usually means the test is skipped, but that only happens if you add `.skip` to the `it` function.

Actually looking into it a bit farther it looks like the shortcut used here won't work on Mac, since on Mac the shortcut is `Cmd+,`
@codedsun Can you make a change to use Command instead of Control for Mac?

**Comment by devinbinnie** [comment]
> @devinbinnie I tried it first, but then the test on the github fails because I think that system is of windows.

@codedsun GitHub is using Linux, but regardless you can check the operating system and make different decisions using `process.platform`

**Comment by codedsun** [comment]
@devinbinnie the doc says command (windows) is accepted in keyTap function while in keyToggle command(mac) is accepted. So keyTap for mac is failing, what you suggest?

http://robotjs.io/docs/syntax#keytapkey-modifier

**Comment by devinbinnie** [comment]
> @devinbinnie the doc says `command (windows)` is accepted in keyTap function while in keyToggle`command(mac)` is accepted. So keyTap for mac is failing, what you suggest?
> 
> http://robotjs.io/docs/syntax#keytapkey-modifier

You can try using `keyToggle` for mac then, see if that works.

**Comment by codedsun** [comment]
@devinbinnie I am confused on what to pass on the down paremeter for `keyToggle(key, down, [modifier])`

**Comment by devinbinnie** [comment]
> @devinbinnie I am confused on what to pass on the down parameter for `keyToggle(key, down, [modifier])`

Looking at the documentation, it looks like you would pass `'down'` or `'up'`. Fiddle with it a bit to see what works for you.

**Comment by codedsun** [comment]
@devinbinnie I tried `null`, `up`, `down`. None of them works, that made me confused on what to put.

**Comment by devinbinnie** [comment]
> @devinbinnie I tried `null`, `up`, `down`, None of them works, that made me confused on what to put.

Perhaps it's a boolean? You could have a look at the RobotJS source code as well.

**Comment by codedsun** [comment]
@devinbinnie https://github.com/octalmage/robotjs/issues/691

**Comment by M-ZubairAhmed** [comment]
@devinbinnie can we make this test so that it can run in linux or windows instead ?

**Comment by codedsun** [comment]
@M-ZubairAhmed - The current test is meant for windows/linux only.

**Comment by devinbinnie** [comment]
@codedsun @M-ZubairAhmed I think we could just skip this test on Mac if it's a hassle to make it work. CI is running Linux so it will be run on every PR anyways.

**Comment by codedsun** [comment]
@devinbinnie @M-ZubairAhmed Review now. Done the changes.

File: e2e/specs/menu_bar/file_menu.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+'use strict';
+
+const fs = require('fs');
+
```
Comment (M-ZubairAhmed) [inline e2e/specs/menu_bar/file_menu.test.js:6]: nit : can we remove this empty line since both of them are external module imports?

**Review by M-ZubairAhmed — APPROVED** [review]
Looks good to me

**Comment by codedsun** [comment]
@M-ZubairAhmed @devinbinnie Done the changes.

File: e2e/specs/menu_bar/file_menu.test.js (hunk 1 above)
Comment (M-ZubairAhmed) [reply e2e/specs/menu_bar/file_menu.test.js]: sorry my bad this changed caused lint error. but i wonder why https://app.circleci.com/pipelines/github/mattermost/desktop/3671/workflows/d7848497-e726-45d7-a644-db615d92b331/jobs/16296?invite=true#step-106-13

Comment (M-ZubairAhmed) [reply e2e/specs/menu_bar/file_menu.test.js:6]: i reverted the change

**Comment by codedsun** [comment]
Thanks a lot @devinbinnie @M-ZubairAhmed 💯

**Comment by BenLloydPearson** [comment]
@codedsun Thanks for your contribution to our desktop contributor event! I wanted to let you know that we've expanded our awards for this event to now include a set of Apple Airpods Pro to our top contributors during the event. You can see more details on [our blog](https://mattermost.com/blog/mattermost-desktop-app-contributor-event/). We'd love to see more contributions for you, and thanks again!

**Pull request merged**